### PR TITLE
adding check for a sync when giving work to miner

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

**Comment by NikVolf** [comment]
https://github.com/ethcore/parity/issues/739

**Comment by gavofyork** [comment]
that's not a valid response. should just not reply.

**Comment by arkpar** [comment]
Syncing status does not take block queue into account queue. Client should expose and additional status or function that also checks if the block queue is empty.

**Comment by NikVolf** [comment]
@gavofyork  As i browsed ethminer code, seems like it would be a valid response
ethminer checks if the first parameter is an empty string

**Comment by gavofyork** [comment]
wasn't part of spec. should be introduced.

File: rpc/src/v1/impls/eth.rs [hunk 1]
```diff
@@ -331,7 +340,7 @@ impl<C, S, A, M, EM> Eth for EthClient<C, S, A, M, EM>
 						let seed_hash = Ethash::get_seedhash(b.block().header().number());
 						to_value(&(pow_hash, seed_hash, target))
 					}
-					_ => Err(Error::invalid_params())
+					_ => Err(Error::internal_error())
```
Comment (gavofyork) [inline rpc/src/v1/impls/eth.rs]: if this is a logic error and cannot possibly happen when the code is bug-free, then it should be a `panic!`

Comment (NikVolf) [inline rpc/src/v1/impls/eth.rs]: it can actually happen, but we have no descriptive errors in rpc
will be adding in next prs

**Pull request merged**